eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -1041,3 +1051,52 @@ constitutionShouldBe cUrl = do
     getsNES $
       nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . constitutionGovStateL
   anchorUrl `shouldBe` fromJust (textToUrl 64 $ T.pack cUrl)
+
+setConstitutionScript :: ConwayEraImp era => ScriptHash (EraCrypto era) -> ImpTestM era ()
+setConstitutionScript sh =
+  modifyNES $
+    nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . constitutionGovStateL
+      .~ Constitution def (SJust sh)
+
+proposalWithGovPolicy ::
+  forall era.
+  ( HasCallStack
+  , ConwayEraImp era
+  ) =>
+  GovAction era ->
+  ScriptHash (EraCrypto era) ->
+  ImpTestM era (ProposalProcedure era)
+proposalWithGovPolicy govAction sh = do
+  rewardAccount <- registerRewardAccount
+  pp <- getsNES $ nesEsL . curPParamsEpochStateL
+  let govAction' = case govAction of
+        ParameterChange gid ppup _ -> ParameterChange gid ppup (SJust sh)
+        TreasuryWithdrawals ws _ -> TreasuryWithdrawals ws (SJust sh)
+        _ -> error "Gov Policy tests are only applicable to ParameterChange and TreasuryWithdrawals actions"
+  pure $
+    ProposalProcedure
+      { pProcReturnAddr = rewardAccount
+      , pProcGovAction = govAction'
+      , pProcDeposit = pp ^. ppGovActionDepositL
+      , pProcAnchor = def
+      }
+
+proposalWithPlutusGovPolicy ::
+  forall era l.
+  ( HasCallStack
+  , ConwayEraImp era
+  , PlutusLanguage l
+  ) =>
+  GovAction era ->
+  Plutus l ->
+  ImpTestM era (ProposalProcedure era)
+proposalWithPlutusGovPolicy govAction plutus = do
+  sh <-
+    impAddPlutusScript $
+      ScriptTestContext
+        plutus
+        PlutusArgs
+          { paSpendDatum = Nothing
+          , paData = P1.I 3
+          }
+  proposalWithGovPolicy govAction sh
```

Comment:
This is not a valid function, since it is not clear will this be a valid proposal or not, which itself will depend on the current constitution. It should be written in the opposite direction. Submit a proposal either with the scripthash that matches the current constitution or with an explicitly mistmatched scripthash. This will make the intent clear at the test site.